### PR TITLE
deprecate `std/threadpool`; use `malebolgia`, `weave`, `nim-taskpool` instead

### PR DESCRIPTION
**Pull request by ringabout**
ref https://github.com/nim-lang/RFCs/issues/503

### CODE REVIEW

**Comment by juancarlospaco** [comment]
I think if we are going to ship 2 package managers with Atlas,
then the wording "Use Nimble X" should be changed to something "package manager agnostic" (?).

File: lib/pure/concurrency/threadpool.nim [hunk 1]
```diff
@@ -7,6 +7,8 @@
 #    distribution, for details about the copyright.
 #
 
+{.deprecated: "use the nimble package `malebolgia` instead".}
```
Comment (ringabout) [inline lib/pure/concurrency/threadpool.nim]:
```suggestion
{.deprecated: "use the nimble package `malebolgia`, `taskpools` or `weave` instead".}
```

File: lib/pure/concurrency/threadpool.nim [hunk 2]
```diff
@@ -7,6 +7,8 @@
 #    distribution, for details about the copyright.
 #
 
+{.deprecated: "use the nimble package `malebolgia`, `taskpools` or `weave` instead".}
```
Comment (ringabout) [inline lib/pure/concurrency/threadpool.nim]:
```suggestion
{.deprecated: "use the nimble packages `malebolgia`, `taskpools` or `weave` instead".}
```

**Comment by metagn** [comment]
"Nimble package" means "package using the Nimble package format", which basically every Nim library is. Both package managers as well as other user package managers recognize the Nimble package format. "Use Nimble" is not necessarily implied.

**Comment by Vindaar** [comment]
Is it wise to recommend people use `weave` at this time? I've had lots of trouble since the switch to ARC/ORC with it unfortunately. Last I remember mratsim talking about it (back when ARC was still a bit off), support for it wasn't really there (at least not tested). Has that changed, @mratsim?

**Comment by mratsim** [comment]
> Is it wise to recommend people use `weave` at this time? I've had lots of trouble since the switch to ARC/ORC with it unfortunately. Last I remember mratsim talking about it (back when ARC was still a bit off), support for it wasn't really there (at least not tested). Has that changed, @mratsim?

No change, I didn't test Weave with arc/orc. And it doesn't call destructors for sure.

I didn't test nim-taskpools either, but at least std/tasks that it relies on does call destructors.

**Comment by juancarlospaco** [comment]
The benefits of moving libs out of std... 😇

**Comment by arnetheduck** [comment]
`std/tasks` is broken in that it allows exceptions to leak out from tasks and their destructors which breaks anything that builds on top of it but fixing it requires a backwards-incompatible change - taskpools no longer uses it for this and other reasons such as memory allocation, alignment needs etc - a type like `Task` has little chance of working well in std until the infrastructure around it exists in a stable form - it could for example be moved to malebolgia.

**Comment by Araq** [comment]
> it could for example be moved to malebolgia.

Malebolgia is likely to use its own version of it, eventually...

**Pull request merged**